namespace/namespace.go

```diff
@@ -39,6 +42,27 @@ func MustNew(version uint8, id []byte) Namespace {
 	return ns
 }
 
+// NewFromBytes returns a new namespace from the provided byte slice.
+func NewFromBytes(bytes []byte) (Namespace, error) {
+	if len(bytes) != NamespaceSize {
+		return Namespace{}, fmt.Errorf("invalid namespace length: %v must be %v", len(bytes), NamespaceSize)
+	}
+
+	err := validateVersionSupported(bytes[0])
+	if err != nil {
+		return Namespace{}, err
+	}
+
+	err = validateID(bytes[0], bytes[1:])
```

Comment:
Looks like validateID taking 2 params is an artifact from the previous implementation, perhaps validateID can take just full `bytes` now instead of subsclices.

Reply:
In `New` we still pass two arguments (ID and Version). `validateID` matches that function signature